plugin/openassetio_manager_bal/bal.py

```diff
@@ -360,20 +359,28 @@ def _copy_and_expand_trait_properties(entity_version_dict: dict, library: dict)
 def normalize_file_url(maybe_file_url):
     """
     Any string in the library that begins with "file:" will be
-    normalized, the main point of this being to collapse relative
-    paths that have been written into the library json.
+    normalized, the main point of this being to collapse upward
+    traversals that have been written into the library json.
     """
+    # Modified from https://stackoverflow.com/a/4317446
 
     # Parse the URL into components
     parsed_url = urlparse(maybe_file_url)
-    path = url2pathname(parsed_url.path)
 
-    # Resolve, removing relative paths, (normalizing)
-    resolved_path = str(pathlib.Path(path).resolve())
+    # Resolve upward traversals (`..`).
+    normed_path = posixpath.normpath(parsed_url.path)
+    if parsed_url.path.endswith("/"):
+        normed_path += "/"
+
+    if parsed_url.path == normed_path:
+        # No changes were made, so avoid unintended mutations by
+        # returning the input unmodified. E.g. `file:123` would get
+        # normalised to `file:///123`, which might not be what we want.
+        return maybe_file_url
```

Comment:
[Question] This a real behaviour issue or just a cover-your-ass defensive programming sort of thing? Presumably `file:///123` would still be valid?

Not saying to remove it, it's probably a good idea in general.

Reply:
My thought was to try to cover cases where it's not actually a file URL and for some reason there's a genuine property with a value of `file:something`, perhaps some kind of tag, that's not a URL.  

There's also the potential case of odd non-conforming applications that have dodgy rules for URLs to work, and normalising to Python standard would break it.

In general it seems a good practice to not alter user's data unless absolutely necessary.

Reply:
I guess there is a counter argument, for consistency and documentation, we can just say "if you use something that looks like a file URL, it _will_ be normalised, deal with it".

Reply:
I tend to agree with the approach tbh, was just wondering if there was a specific issue we were working around or not.